docs/design/autoware-interfaces/components/vehicle-dimensions.md

```diff
@@ -0,0 +1,68 @@
+# Vehicle dimensions
+
+## Vehicle axes and base_link
+
+![Vehicle Axes](images/vehicle_axes.svg){: style="width:500px"}
+
+`base_link` frame is in the center of the rear axle.
+`base_link` used very frequently throughout the stack.
+
+- Localization module outputs the `map` to `base_link` transformation.
+- Planning module plans the poses for where the `base_link` frame should be in the future.
+- Control module tries to fit `base_link` to incoming poses.
+
+## Vehicle dimensions
+
+![Vehicle Dimensions](images/vehicle_dimensions.svg){: style="width:750px"}
+
+### wheelbase
+
+The distance between front and rear axles.
+
+### track_width
```

Comment:
Will you add this and some other variables to `vehicle_info`?
https://github.com/autowarefoundation/autoware.universe/blob/c555767cea9b1d75217597f6cd3a0b2faefe9439/vehicle/vehicle_info_util/include/vehicle_info_util/vehicle_info.hpp

Reply:
Some comments :pray: 
- Currently it is wheel_tread, not track_width.
- I don't know with whether to include mirrors in left, right_overhang. Currently it is ambiguous.
- I would also like a description of the tires. Mainly used for dead reckoning.

Reply:
@kenji-miyake 

So, I've ported `vehicle_info` to Autoware.Auto before: https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/merge_requests/1150/diffs#4054286b653c00d4ad6fddafde60ef1acc5a14a0 and I'm familiar with it.

I've looked for how anyone else names the dist between left and right wheel centers:
- https://en.wikipedia.org/wiki/Axle_track
- https://itstillruns.com/measure-track-vehicle-5850001.html
- https://www.sodialed.com/rc-setup-tips/track-width
And found no results for wheel tread. So I think we should rename `wheel_tread` to `track_width` in the vehicle_info and anything that uses it.

@yukkysaito said:
> I don't know with whether to include mirrors in left, right_overhang. Currently it is ambiguous.

I'm guessing overhangs are only for collider box/prism of the vehicle for obstacle avoidance.

If so, it is better to include it for the minimum bounding box/prism.

If not, could you explain why it exists?

@yukkysaito said:
> I would also like a description of the tires. Mainly used for dead reckoning.

I can add a small illustration for it too, thanks.

Reply:
@xmfcx 
>I'm guessing overhangs are only for collider box/prism of the vehicle for obstacle avoidance.
If so, it is better to include windows and any other protrusions for the minimum bounding box/prism.
If not, could you explain why it exists?

The left and right overhangs are used to determine collision and to filter the point clouds hitting the ego vehicle.
There are two issues when mirror is included.
- The size of the vehicle will be larger than the actual vehicle size, making it impossible to drive on narrow roads. This has a significant impact on long vehicles such as buses.
- It would not be able to detect people near the vehicle. (nitpicky)

There are two issues when mirror is not included.
- might hit the mirror of the car.
- The point cloud hits the mirror and there are obstacles near the ego vehicle.


The current implementation does not include mirror, but has a special mirror yaml to filter the point cloud.
https://github.com/autowarefoundation/sample_vehicle_launch/blob/main/sample_vehicle_description/config/mirror.param.yaml
However, there is a possibility of hitting the mirror.

Reply:
![image (1)](https://user-images.githubusercontent.com/8327598/162957931-faa9e48b-133f-40e1-8857-1113a00150bb.png)
There are also mirrors like this that are asymmetrical and very long on one side.
This make us worry :cry:

Reply:
@xmfcx update with the polygon footprint around the vehicle.